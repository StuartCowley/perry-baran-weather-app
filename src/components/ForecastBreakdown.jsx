import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import getUnits from "../helpers/getUnits";
import useUnitContext from "../hooks/useUnitContext";
import "../styles/ForecastBreakdown.css";

function ForecastBreakdown({
  icon,
  temp,
  minTemp,
  maxTemp,
  humidity,
  windSpeed,
  weather,
}) {
  const { units } = useUnitContext();
  const { tempUnits, speedUnits } = getUnits(units);

  return (
    <div className="forecast-breakdown">
      <div className="forecast-breakdown__right">
        <div data-testid="forecast-icon">
          <WeatherIcon name="owm" iconId={icon} />
        </div>
        <p>{weather}</p>
      </div>
      <div className="forecast-breakdown__left">
        <p className="forecast-breakdown__temp">
          {temp}
          <sup>{tempUnits}</sup>
        </p>
        <div>
          <h3>Min: </h3>
          <p>
            {minTemp}
            {tempUnits}
          </p>
        </div>
        <div>
          <h3>Max: </h3>
          <p>
            {maxTemp}
            {tempUnits}
          </p>
        </div>
        <div>
          <h3>Humidity: </h3>
          <p>{humidity}%</p>
        </div>
        <div>
          <h3>Wind Speed: </h3>
          <p>
            {windSpeed}
            {speedUnits}
          </p>
        </div>
      </div>
    </div>
  );
}

ForecastBreakdown.propTypes = {
  icon: PropTypes.string.isRequired,
  temp: PropTypes.string.isRequired,
  minTemp: PropTypes.string.isRequired,
  maxTemp: PropTypes.string.isRequired,
  humidity: PropTypes.string.isRequired,
  windSpeed: PropTypes.string.isRequired,
  weather: PropTypes.string.isRequired,
};

export default ForecastBreakdown;
