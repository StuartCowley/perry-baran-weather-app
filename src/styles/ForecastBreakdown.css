.forecast-breakdown {
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
  flex: 1 1 auto;
  text-align: center;
  min-height: 350px;
  padding: 20px;
}

.forecast-breakdown__left {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.forecast-breakdown__left h3, .forecast-breakdown__left h3 + p {
  display: inline;
}

.forecast-breakdown__left > div > * {
  font-size: min(1.3rem, 3vw);
}

.forecast-breakdown__temp {
  font-size: min(7rem, 15vw);
}

.forecast-breakdown__right i {
  font-size: min(13rem, 18vw);
}

.forecast-breakdown__right p {
  font-size: min(2rem, 2.5vw);
  font-weight: bold;
  color: white;
  text-shadow: var(--text-shadow-black);
  position: relative;
  top: min(40px, 4vw);
}