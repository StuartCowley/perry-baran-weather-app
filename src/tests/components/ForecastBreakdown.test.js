import React from "react";
import { render } from "@testing-library/react";
import ForecastBreakdown from "../../components/ForecastBreakdown";

describe("ForecastBreakdown", () => {
  const validProps = {
    icon: "200",
    temp: "20",
    minTemp: "18",
    maxTemp: "22",
    humidity: "30",
    windSpeed: "40",
    weather: "cold",
  };
  const { icon, temp, minTemp, maxTemp, humidity, windSpeed, weather } =
    validProps;
  let screen;

  beforeEach(() => {
    screen = render(
      <ForecastBreakdown
        icon={icon}
        temp={temp}
        minTemp={minTemp}
        maxTemp={maxTemp}
        humidity={humidity}
        windSpeed={windSpeed}
        weather={weather}
      />
    );
  });

  test("snapshot", () => {
    const { asFragment } = screen;

    expect(asFragment()).toMatchSnapshot();
  });

  test("correctly renders props", () => {
    const { getByText, getByTestId } = screen;

    expect(getByTestId("forecast-icon")).toContainHTML("i");
    expect(getByText(new RegExp(temp))).toBeInstanceOf(HTMLParagraphElement);
    expect(getByText(new RegExp(minTemp))).toBeInstanceOf(HTMLParagraphElement);
    expect(getByText(new RegExp(maxTemp))).toBeInstanceOf(HTMLParagraphElement);
    expect(getByText(new RegExp(humidity))).toBeInstanceOf(
      HTMLParagraphElement
    );
    expect(getByText(new RegExp(windSpeed))).toBeInstanceOf(
      HTMLParagraphElement
    );
  });

  test("it contains correct headings", () => {
    const { getByText } = screen;

    expect(getByText(/min:/i)).toBeInstanceOf(HTMLHeadingElement);
    expect(getByText(/max:/i)).toBeInstanceOf(HTMLHeadingElement);
    expect(getByText(/humidity:/i)).toBeInstanceOf(HTMLHeadingElement);
    expect(getByText(/wind speed:/i)).toBeInstanceOf(HTMLHeadingElement);
  });
});
