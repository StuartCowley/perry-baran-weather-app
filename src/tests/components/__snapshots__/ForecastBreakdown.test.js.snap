// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastBreakdown snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-breakdown"
  >
    <div
      class="forecast-breakdown__right"
    >
      <div
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-storm-showers"
          name="owm"
        />
      </div>
      <p>
        cold
      </p>
    </div>
    <div
      class="forecast-breakdown__left"
    >
      <p
        class="forecast-breakdown__temp"
      >
        20
        <sup>
          °C
        </sup>
      </p>
      <div>
        <h3>
          Min: 
        </h3>
        <p>
          18°C
        </p>
      </div>
      <div>
        <h3>
          Max: 
        </h3>
        <p>
          22°C
        </p>
      </div>
      <div>
        <h3>
          Humidity: 
        </h3>
        <p>
          30%
        </p>
      </div>
      <div>
        <h3>
          Wind Speed: 
        </h3>
        <p>
          40m/s
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
